lib/webinspector/transformer/webinspector-encoder.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable promise/prefer-await-to-callbacks */
+import Stream from 'stream';
+import { createPlist } from '../../util/plist';
+
+const WEBINSPECTOR_PARTIAL_PACKET_CHUNK_SIZE = 8096;
+
+class WebInspectorEncoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+  }
+
+  _transform (data, encoding, callback) {
+    let payloadBuffer = createPlist(data, true);
+    for (let i = 0; i < payloadBuffer.length; i += this._encode(payloadBuffer, 0)) {}
```

Comment:
for(); should be enough

Reply:
I actually have a bug there. It should be `this._encode(payloadBuffer, i)`